### PR TITLE
[IMP] sale_(project,timesheet): add tests to project apps

### PR DESCRIPTION
**Pull request by dise-odoo**
This PR will add various tests to all projects app.

task-3231698

### CODE REVIEW

**Review by auon-odoo — CHANGES_REQUESTED** [review]

File: addons/project/tests/test_project_subtasks.py [hunk 1]
```diff
@@ -354,3 +354,12 @@ def test_get_all_subtasks(self):
             self.assertEqual(parent_id, task.id, "The key should be the common ancestor")
             self.assertEqual(set(subtask_ids), set(all_subtasks.ids),
                              "All subtasks linked to the common ancestor should be returned by _get_subtask_ids_per_task_id method")
+
+    def test_subtask_copy_followers(self):
+        """ This test will check that a task will propagate its partners to its subtasks """
+        task_form = Form(self.task_1.with_context({'tracking_disable': True}))
```
[the remark this answers is not in the document]
Comment (dise-odoo) [reply addons/project/tests/test_project_subtasks.py:362]: oop right, I edited this test ;P

**Review by auon-odoo — APPROVED** [review]
lgtm, thanks!!

**Review by MissingNoShiny — COMMENTED** [review]
Thanks for your work, and especially for creating each test in a different commit :slightly_smiling_face: 
I just have two small comments

File: addons/project/tests/test_project_config.py [hunk 2]
```diff
@@ -57,3 +57,12 @@ def test_new_projects_enable_features(self):
...
                 f"{config_flag}/{project_flag} feature")
+
+    def test_project_stages_feature_enable_views(self):
+        """Check that the Gantt, Calendar and Activities views are
+        enabled when the 'Project Stage' feature is enabled.
+        """
+        self.Settings.create({"group_project_stages": True}).execute() # enabling feature
+        menu_ids = set([self.env.ref('project.menu_projects').id, self.env.ref('project.menu_projects_config').id])
+        menu_loaded = set(self.env['ir.ui.menu']._load_menus_blacklist())
+        self.assertTrue(menu_ids.issubset(menu_loaded), "The menu project and menu projects config should be loaded")
```
[the remark this answers is not in the document]
Comment (dise-odoo) [reply addons/project/tests/test_project_config.py:68]: It's Xavier's knowledge, not mine :P

File: addons/project/tests/test_task_recurrence.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from .test_project_base import TestProjectCommon
+
+class TestTaskRecurrence(TestProjectCommon):
```
[the remark this answers is not in the document]
Comment (dise-odoo) [reply addons/project/tests/test_task_recurrence.py:3]: Indeed, and [this test](https://github.com/odoo/odoo/blob/17.0/addons/project/tests/test_project_recurrence.py#L51) is mostly doing what I'm doing. I will simply edit it to add the `user_ids` field.

**Review by MissingNoShiny — APPROVED** [review]

**Review by xavierbol — APPROVED** [review]
LGTM, nice work! 💪🏼 

robodoo rebase-ff r+

**Pull request closed without merging**